libs/encryption-sdk/src/main/java/org/opensearch/encryption/frame/core/EncryptionHandler.java

```diff
@@ -0,0 +1,326 @@
+import com.amazonaws.encryptionsdk.model.CiphertextType;
+import com.amazonaws.encryptionsdk.model.KeyBlob;
+import org.bouncycastle.asn1.ASN1Encodable;
+import org.bouncycastle.asn1.ASN1Integer;
+import org.bouncycastle.asn1.ASN1Sequence;
+import org.bouncycastle.asn1.DERSequence;
+
+import javax.crypto.SecretKey;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.interfaces.ECPrivateKey;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class implements the CryptoHandler interface by providing methods for the encryption of
+ * plaintext data.
+ *
+ * <p>This class creates the ciphertext headers and delegates the encryption of the plaintext to the
+ * {@link FrameEncryptionHandler}.
+ */
+public class EncryptionHandler implements MessageCryptoHandler {
+
+    private final Map<String, String> encryptionContext_;
+    private final CryptoAlgorithm cryptoAlgo_;
+    private final List<MasterKey> masterKeys_;
+    private final List<KeyBlob> keyBlobs_;
+    private final SecretKey encryptionKey_;
+    private final byte version_;
+    private final CiphertextType type_;
+    private final byte nonceLen_;
+
+    private final CiphertextHeaders ciphertextHeaders_;
+    private final byte[] ciphertextHeaderBytes_;
+    private final CryptoHandler contentCryptoHandler_;
+
+    private boolean firstOperation_;
+    private boolean complete_ = false;
+
+    private long plaintextBytes_ = 0;
+    private long plaintextByteLimit_ = -1;
+
+    private final PrivateKey trailingSignaturePrivateKey;
+    private final MessageDigest trailingDigest;
+    private final Signature trailingSig;
+
+    /**
+     * Create an encryption handler using the provided master key and encryption context.
+     * @param encryptionMetadata Context object created before encryption
+     * @param isFirstStream In case of first stream, file header is additionally created which consists of crypto
+     *                      materials.
+     * @param frameStartNumber Number from which assignment has to start for new frames
+     */
+    public EncryptionHandler(EncryptionMetadata encryptionMetadata, boolean isFirstStream, int frameStartNumber) throws AwsCryptoException {
+        encryptionContext_ = encryptionMetadata.getEncryptionContext();
+        cryptoAlgo_ = encryptionMetadata.getCryptoAlgo();
+        masterKeys_ = encryptionMetadata.getMasterKeys();
+        keyBlobs_ = encryptionMetadata.getKeyBlobs();
+        encryptionKey_ = encryptionMetadata.getEncryptionKey();
+        version_ = encryptionMetadata.getVersion();
+        type_ = encryptionMetadata.getType();
+        nonceLen_ = encryptionMetadata.getNonceLen();
+        ciphertextHeaders_ = encryptionMetadata.getCiphertextHeaders();
+        ciphertextHeaderBytes_ = encryptionMetadata.getCiphertextHeaderBytes();
```

Comment:
I kept metadata attributes only instead of encryptionMetadata to allow handler to be created even with ParsedCiphertext which contains same info but is a different object type. One use case could be where we have to re-encrypt some content. In this use case, we will have a ParsedCipherText and will need to create encryption handler for encryption. We don't have this use case yet though.